contracts/utils/DAOTracker.sol

```diff
@@ -0,0 +1,90 @@
+pragma solidity ^0.5.8;
+
+import "@daostack/infra/contracts/Reputation.sol";
+import "../controller/DAOToken.sol";
+import "../controller/Avatar.sol";
+import "../controller/ControllerInterface.sol";
+import "openzeppelin-solidity/contracts/ownership/Ownable.sol";
+
+/**
+ * @title An on-chain "source of truth" for what DAOs
+ *        should be index into DAOstack's subgraph.
+ */
+contract DAOTracker is Ownable {
+
+    // `blacklist` the DAO from the subgraph's cache.
+    // Only able to be set by the owner of the DAOTracker.
+    mapping(address=>bool) public blacklisted;
+
+    event TrackDAO(address indexed _avatar, address _controller, address _reputation, address _daoToken);
+    event BlacklistDAO(address indexed _avatar, string _explanationHash);
+    event ResetDAO(address indexed _avatar, string _explanationHash);
+
+    modifier onlyAvatarOwner(Avatar avatar) {
+        require(avatar.owner() == msg.sender,
+                "The caller must be the owner of the Avatar.");
+        _;
+    }
+
+    modifier notBlacklisted(Avatar avatar) {
+        require(blacklisted[address(avatar)] == false,
+                "The avatar has been blacklisted.");
+        _;
+    }
+
+    /**
+    * @dev track a new organization. This function will tell the subgraph
+    *      to start ingesting events from the DAO's contracts.
+    *      NOTE: This function should be called as early as possible in the DAO deployment
+    *      process. **Smart Contract Events that are emitted from blocks prior to this function's
+    *      event being emitted WILL NOT be ingested into the subgraph**, leading to an incorrect
+    *      cache. If this happens to you, please contact the subgraph maintainer. Your DAO will
+    *      need to be added to the subgraph's startup config, and the cache will need to be rebuilt.
+    * @param _avatar the organization avatar
+    * @param _controller the organization controller
+    */
+    function track(Avatar _avatar, ControllerInterface _controller)
```

Comment:
it seems to be safer and cleaner to pass in only avatar and get the controller as the avatar owner.

Reply:
The reason for not doing this is because I wanted to make sure the `track(...)` function was being called by the current owner of the Avatar, and not some random person / caller. See modifier `onlyAvatarOwner`. I saw this as a good safety precaution that would ensure DAOs that don't want to be indexed, couldn't be.

Reply:
though in this case a dao which is created by daoCreator.sol will be index .want it or not .
dao which would like not to be index ,(not sure why he wants that), will be needed to be deployed manually and loose the advantage of using daoCreator.sol .

if we are convinced ,which I am still not, that there is a dao which cares if it is indexed or not , 
we can support that by passing a parameter to the daoCreator.sol forge func - bool _index ?

Reply:
I suppose I misspoke when I said "...a good safety precaution that would ensure DAOs that don't want to be indexed, couldn't be".

What I really meant to say was "I saw this as a good safety precaution that would ensure the DAO being tracked was in the process of being constructed, and wasn't already live".

I see both as benefits to ensuring msg.sender is avatar.owner(), would you agree?

Reply:
make sense 👍